### PR TITLE
Eliminate `causer` interface in favour of errors.Unwrap

### LINKED ISSUE #6211: Eliminate `causer` interface in favor of `errors.Unwrap`
**How to categorize this issue?**

/area dev-productivity
/kind cleanup technical-debt

**What would you like to be added**:
The `causer` interface https://github.com/gardener/gardener/blob/862c77153f92be4f9ab8d039d2751f630539ef90/pkg/utils/errors/unwrap.go#L19-L21 leads to confusion when it comes to error handling and error propagation.
Let's eliminate it and rather use Golang's native error unwrapping so that this code becomes more clear.

### PR DESCRIPTION
**Pull request by ary1992**
**How to categorize this PR?**

/area dev-productivity
/kind cleanup technical-debt

**What this PR does / why we need it**:
Eliminate caused interface and rather use Golang`s native error unwrapping. Ref : https://pkg.go.dev/errors
**Which issue(s) this PR fixes**:
Fixes https://github.com/gardener/gardener/issues/6211

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Comment by plkokanov** [comment]
There is one more usage of the causer interface in the retry package: https://github.com/gardener/gardener/blob/eef61292eb6d9abbce877f11c7179fa491897ecb/pkg/utils/retry/retry.go#L117
Should we also get rid of that one?

**Review by plkokanov — CHANGES_REQUESTED** [review]
With @shafeeqes we found a few more of our errors that use the `causer` interface:
https://github.com/gardener/gardener/blob/0594d5fb4bdc34c5711b72f106b1eb822a4ac620/pkg/utils/flow/flow.go#L382
https://github.com/gardener/gardener/blob/0594d5fb4bdc34c5711b72f106b1eb822a4ac620/pkg/utils/flow/flow.go#L390

File: pkg/utils/retry/retry.go [hunk 1]
```diff
@@ -113,14 +113,6 @@ type Error struct {
 	err      error
 }
 
-// Cause implements Causer.
-func (e *Error) Cause() error {
-	if e.err != nil {
-		return e.err
-	}
-	return e.ctxError
```
Comment (plkokanov) [inline pkg/utils/retry/retry.go:121]: I think you can still return the context error in the `Unwrap()` method.

**Review by plkokanov — CHANGES_REQUESTED** [review]

**Comment by shafeeqes** [comment]
Can you please adapt this doc string as well?
https://github.com/gardener/gardener/blob/2f6567da34b2d70ef8288e9c0cb10fcce19c35da/pkg/utils/errors/errors.go#L85

**Comment by shafeeqes** [comment]
I don't have any other comments.

File: pkg/utils/errors/unwrap.go [hunk 2]
```diff
@@ -16,30 +16,18 @@ package errors
...
 
-type causer interface {
-	Cause() error
-}
-
-// Unwrap unwraps and returns the root error. Multiple wrappings either via `fmt.Errorf` or via `causer` implementations are properly taken into account.
+// Unwrap unwraps and returns the root error. Multiple wrappings via `fmt.Errorf` implementations are properly taken into account.
 func Unwrap(err error) error {
-	var cdone, udone bool
+	var udone bool
```
Comment (plkokanov) [inline pkg/utils/errors/unwrap.go]: nit: can you rename `udone` variable, there's no reason for the weird name now.
```suggestion
	var done bool
```

**Review by plkokanov — CHANGES_REQUESTED** [review]
I ran a couple of quick tests to check if the flow and retry errors that are reported remain the same (before and after this PR) and that seems to be the case.

File: pkg/utils/errors/unwrap.go (hunk 2 above)
Comment (ary1992) [reply pkg/utils/errors/unwrap.go]: Done.

**Pull request merged**